spark/v3.2/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestMerge.java

```diff
@@ -84,17 +86,22 @@ public void testMergeWithStaticPredicatePushDown() {
 
     sql("ALTER TABLE %s ADD PARTITION FIELD dep", tableName);
 
-    append(tableName,
-        "{ \"id\": 1, \"dep\": \"software\" }\n" +
-        "{ \"id\": 11, \"dep\": \"software\" }\n" +
-        "{ \"id\": 1, \"dep\": \"hr\" }");
+    // add a data file to the 'software' partition
+    append(tableName, "{ \"id\": 1, \"dep\": \"software\" }");
+
+    // add a data file to the 'hr' partition
+    append(tableName, "{ \"id\": 1, \"dep\": \"hr\" }");
 
     Table table = validationCatalog.loadTable(tableIdent);
 
     Snapshot snapshot = table.currentSnapshot();
     String dataFilesCount = snapshot.summary().get(SnapshotSummary.TOTAL_DATA_FILES_PROP);
     Assert.assertEquals("Must have 2 files before MERGE", "2", dataFilesCount);
 
+    // remove the data file from the 'hr' partition to ensure it is not scanned
+    DataFile dataFile = Iterables.getOnlyElement(snapshot.addedFiles());
+    table.io().deleteFile(dataFile.path().toString());
```

Comment:
Could you move it and move it back so that you don't remove the validation at the end of the test? Something like...

```java
  withFilesUnavailable(Iterables.getOnlyElement(snapshot.addedFiles()).path().toString()) {
    createOrReplaceView(...);
    withSQLConf(...) { sql("MERGE INTO ...") };
  }
```

You could move the files to a temp location, then copy them back at the end.

Reply:
Yeah, this will be helpful for future use cases. Let me spend some time on this one now.

Reply:
Fixed.